modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/lang/files/Open.java

```diff
@@ -0,0 +1,97 @@
+package org.ballerinalang.nativeimpl.lang.files;
+
+
+import org.ballerinalang.bre.Context;
+import org.ballerinalang.model.types.TypeEnum;
+import org.ballerinalang.model.values.BString;
+import org.ballerinalang.model.values.BStruct;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.natives.AbstractNativeFunction;
+import org.ballerinalang.natives.annotations.Argument;
+import org.ballerinalang.natives.annotations.Attribute;
+import org.ballerinalang.natives.annotations.BallerinaAnnotation;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.util.exceptions.BallerinaException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.util.Locale;
+
+/**
+ * Gets the streams from a local file.
+ */
+@BallerinaFunction(
+        packageName = "ballerina.lang.files",
+        functionName = "open",
+        args = {@Argument(name = "file", type = TypeEnum.STRUCT, structType = "File",
+                structPackage = "ballerina.lang.files"), @Argument(name = "accessMode", type = TypeEnum.STRING)},
+        isPublic = true
+)
+@BallerinaAnnotation(annotationName = "Description", attributes = { @Attribute(name = "value",
+        value = "Gets streams from a local file") })
+@BallerinaAnnotation(annotationName = "Param", attributes = {@Attribute(name = "file",
+        value = "The File that should be opened") })
+@BallerinaAnnotation(annotationName = "Param", attributes = {@Attribute(name = "accessMode",
+        value = "The mode the file should be opened in") })
+public class Open extends AbstractNativeFunction {
+
+    private static final Logger log = LoggerFactory.getLogger(Open.class);
+    @Override public BValue[] execute(Context context) {
+        BStruct struct = (BStruct) getArgument(context, 0);
+        BString accessMode = (BString) getArgument(context, 1);
+        try {
+            String path = struct.getValue(0).stringValue();
+            File file = new File(path);
+            String accessLC = accessMode.stringValue().toLowerCase(Locale.getDefault());
+            if (accessLC.contains("r")) {
+                BufferedInputStream is = new BufferedInputStream(new FileInputStream(file));
+                struct.addNativeData("inStream", is);
+            }
+            if (accessLC.contains("a") && accessLC.contains("w")) {
+                log.info("found both 'a' and 'w' in access mode string, opening file in append mode");
+            }
+            if (accessLC.contains("a")) {
+                createDirs(file);
+                BufferedOutputStream os = new BufferedOutputStream(new FileOutputStream(file));
+                struct.addNativeData("outStream", os);
+            } else if (accessLC.contains("w")) {
+                createDirs(file);
```

Comment:
Need to check for `file.exists()` and `file.canWrite()`.
if the file already exists, print a warning (?)
If the file cannot be written, throw an error with a proper message.
Same before opening for reading, and appending.